sled-agent/src/bootstrap/agent.rs

```diff
@@ -76,6 +88,151 @@ impl From<BootstrapError> for ExternalError {
     }
 }
 
+// Represents the bootstrap agent's view into hardware.
+//
+// The bootstrap agent is responsible for launching the switch zone, which it
+// can do once it has detected that the corresponding driver has loaded.
+struct HardwareMonitorWorker {
+    log: Logger,
+    exit_rx: oneshot::Receiver<()>,
+    hardware: HardwareManager,
+    services: ServiceManager,
+}
+
+impl HardwareMonitorWorker {
+    fn new(
+        log: Logger,
+        exit_rx: oneshot::Receiver<()>,
+        hardware: HardwareManager,
+        services: ServiceManager,
+    ) -> Self {
+        Self { log, exit_rx, hardware, services }
+    }
+
+    async fn run(
+        mut self,
+    ) -> Result<(HardwareManager, ServiceManager), BootstrapError> {
```

Comment:
A very similar type of worker exists in `sled_agent.rs`, which also monitors for the switch zone.

However, these "workers" are distinct -- the bootstrap agent version *only* looks for the switch zone, but the sled agent one is presumably going to monitor for a lot more signals from hardware.

Additionally, the bootstrap agent is activating the switch zone without an underlay IP address, while the sled agent *always* activates the switch zone with an underlay IP address.